File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {   
    console.log("기사단", number)
    
    console.log("제한", limit)
    
    console.log("공격력", power)
    
    const knights = [];
    
    // 약수의 개수 
    for(let i = 1; i <= number; i++){
        let divisors =0;
        
        for(let j = 1; j <= Math.floor(Math.sqrt(i)); j++){
           if(i % j === 0) {
               if(i / j ===  j) divisors++
               else divisors += 2;
                   
           }
        }
        
        knights.push(divisors)
    }
    
    const sum = knights.reduce((acc, cur) => {
        if(cur > limit) return acc + power;
        else return acc + cur
    }, 0)
    
    return sum;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
import math

def solution(number, limit, power):
    total = 0
    
    for num in range(1, number +1):
        curPow = countDivisors(num)
        
        if curPow > limit:
            total += power
        else:
            total += curPow
    
    return total
        
def countDivisors (number):
        if number == 1:
            return 1
        
        counts = 0
        for i in range(1, math.isqrt(number) + 1):
            if number % i == 0:
                if number // i == i:
                    counts += 1
                else:
                    counts += 2
        return counts